### PR TITLE
Add `src/components/MapControlGeocoding.js` and `src/components/MapControlNominatimResults.js`

### PR DESCRIPTION
**Pull request by Raruto**
## List of changes

- refactor `src/app/g3w-ol/controls/geocodingcontrol.js` and
  - extract functional component into `src/components/MapControlGeocoding.js`
  - extract functional component into `src/components/MapControlNominatimResults.js`

Related to https://github.com/g3w-suite/g3w-client/issues/193 and preparatory for https://github.com/g3w-suite/g3w-client/issues/85

### CODE REVIEW

File: src/app/g3w-ol/controls/geocodingcontrol.js [hunk 1]
```diff
@@ -400,6 +404,7 @@ function GeocodingControl(options={}) {
     debug: false,
     viewbox: options.bbox,
     bounded: 1,
+    classMobile: options.isMobile ? 'nominatim-mobile' : '',
```
Comment (volterra79) [inline src/app/g3w-ol/controls/geocodingcontrol.js]: @Raruto where is defined this class? I can't find it in previous code

**Review by volterra79 — APPROVED** [review]

File: src/app/g3w-ol/controls/geocodingcontrol.js (hunk 1 above)
Comment (Raruto) [reply src/app/g3w-ol/controls/geocodingcontrol.js]: @volterra79 could it have been removed in here? https://github.com/g3w-suite/g3w-client/commit/dd04b597da76c5deb60a456171458958a8ff6472

For more info 👉 https://github.com/g3w-suite/g3w-client/commits/dev/src/app/g3w-ol/controls/geocodingcontrol.js

Comment (Raruto) [inline src/app/g3w-ol/controls/geocodingcontrol.js]:
```suggestion
```

File: src/app/g3w-ol/controls/geocodingcontrol.js [hunk 2]
```diff
@@ -445,76 +450,13 @@ function GeocodingControl(options={}) {
...
-        ]),
-        h('ul', {
-          class: {
-            [cssClasses.inputText.result]: true
-          }
-        })])
-    }
-  });
+  const GeocoderVueContainer = Vue.extend(MapControlGeocoding({
+    containerClass: `${cssClasses.namespace} ${cssClasses.inputText.container} ${this.options.classMobile}`,
```
Comment (Raruto) [inline src/app/g3w-ol/controls/geocodingcontrol.js]:
```suggestion
    containerClass: `${cssClasses.namespace} ${cssClasses.inputText.container}`,
```

**Review by volterra79 — APPROVED** [review]

**Pull request merged**